webapp/web-components/cht-form/src/app.component.ts

```diff
@@ -236,15 +247,36 @@ export class AppComponent {
 
   private tearDownForm() {
     this.unloadForm();
-    this._formXml = undefined;
-    this._formHtml = undefined;
-    this._formModel = undefined;
-    this._contactSummary = undefined;
-    this._contactType = undefined;
-    this._content = null;
-    this._formId = this.DEFAULT_FORM_ID;
-    this._user = this.DEFAULT_USER;
+
     this.editing = false;
     this.status = { ...this.DEFAULT_STATUS };
+    this.currentRender = undefined;
+    this.reRenderForm = false;
+
+    // The web component framework does some kind of "lazy setting" where it will not call a setter with the same value
+    // twice. So, we cannot just reset the internal state of the "inputs" here. We need to call "through the front door"
+    //  so the context knows that the values have changed instead of just directly resetting the state of this class.
+    const myForm = this.document
+      .getElementById(this._formId)
+      ?.closest('cht-form');
+    const component = myForm || this;
+    // @ts-expect-error it does exist
+    component.formXml = undefined;
+    // @ts-expect-error it does exist
+    component.formHtml = undefined;
+    // @ts-expect-error it does exist
+    component.formModel = undefined;
+    // @ts-expect-error it does exist
+    component.contactSummary = undefined;
```

Comment:
I have a PR that adds an additional field to formContext. This link between form-context AND the cht-form sems very loose. I wouldn't know I had to reset this new field if I had not read this code. 

Is it possible to somehow iterate over the possible properties of the formContext interface, to get their names, and reset their corresponding value in this component instead of hardcoding them here as well?

Reply:
In theory I think we could make `EnketoFormContext` an interface and then include an implementation here in cht-form.  Then, when the interface changes, the cht-form build should break if its data model is not updated with the new interface changes.  I do kind of like this idea!  :thinking:

Reply:
If something forces the two types to be synced, I'm happy!